tests/utils/test_date.py

```
@@ -253,6 +254,18 @@ def test_overlapping_periods_with_datetime_objects(self, sample_transactions_tab
         ):
             filter_and_label_by_periods(sample_transactions_table, period_ranges)
 
+    def test_with_custom_column_names(self, sample_transactions_table):
+        """Test filter_and_label_by_periods with custom column names."""
+        custom_table = sample_transactions_table.rename({"transaction_date": "my_date_col"})
+        period_ranges = {"Test_Period": ("2023-01-01", "2023-12-31")}
+
+        with option_context("column.transaction_date", custom_date_column):
```

Ruff (0.11.9)

Undefined name `custom_date_column`

(F821)